packages/investor-foxy/src/api/api.ts

```
@@ -805,6 +975,103 @@ export class FoxyApi {
     }
   }
 
+  async getTokeRewardAmount(input: ContractAddressInput): Promise<GetTokeRewardAmount> {
+    const { contractAddress } = input
+    const rewardHashContract = new this.web3.eth.Contract(tokeRewardHashAbi, tokeRewardHashAddress)
+    const latestCycleIndex = await (async () => {
+      try {
+        return rewardHashContract.methods.latestCycleIndex().call()
+      } catch (e) {
+        throw new Error(`Failed to get latestCycleIndex, ${e}`)
+      }
+    })()
+    const cycleHashes = await (async () => {
+      try {
+        return rewardHashContract.methods.cycleHashes(latestCycleIndex).call()
+      } catch (e) {
+        throw new Error(`Failed to get latestCycleIndex, ${e}`)
+      }
+    })()
+
+    try {
+      const { latestClaimable } = cycleHashes
+      const response = await fetch(
+        `${TOKE_IPFS_URL}/${latestClaimable}/${contractAddress.toLowerCase()}.json`
+      )
+      const { payload, signature } = await response.json()
+
+      const claimAmount = bnOrZero(payload.amount)
+      const v = signature.v
+      const r = signature.r
+      const s = signature.s
+      return {
+        latestCycleIndex,
+        claimAmount,
+        v,
+        r,
+        s
+      }
+    } catch (e) {
+      throw new Error(`Failed to get information from Tokemak ipfs ${e}`)
+    }
+  }
+
+  async claimFromTokemak(input: TxInput): Promise<string> {
+    const { contractAddress, wallet, userAddress, accountNumber = 0, dryRun = false } = input
+    this.verifyAddresses([contractAddress])
+
+    const { latestCycleIndex, claimAmount, v, r, s } = await this.getTokeRewardAmount(input)
+
+    if (!wallet) throw new Error('Missing inputs')
```

move this up to 1020 and fail faster